### DISCUSSION TITLE
Testing heterogenous multi device architecture with ZTEST and Twister

### DISCUSSION THREAD

**Discussion by kmeinhar** [zephyrproject-rtos/zephyr#53108] [Q&A]
Hello All,

I am currently developing a board that contains multiple chips that all run Zephyr RTOS. They communicate over UART with each other.
An application could run on MCU A and requests data from MCU B. B then answers and A analyses this data.

I would like to execute Unit Test with ZTEST and Twister where multiple devices are flashed in a specific order and then executed.
As normal Twister would analyze the test results from both devices. 
Is something like this possible? If yes is there already such a test suite in Zephyr? I was not able to find one.

Best Regards,
Kai

**Comment by jcpi-q** [comment]
Not sure this thread will go anywhere but I'm also interested in extending twister to run tests (ztest, pytest at least) on multiple devices connected to each other (physical or radio connections shouldn't make a difference).

Would there be any effort ongoing in that sense already?
